### ISSUE TITLE
conversations table is never cleaned

### ISSUE THREAD

**Issue by BenLubar** [mastodon/mastodon#16629]
### Expected behaviour

When statuses are deleted, empty conversations should not be left behind

### Actual behaviour

I have 7.6 million empty conversations in my database and only 1.5 million conversations with at least one status

### Steps to reproduce the problem

1. run an instance for a while
2. check the result of:
    ```sql
    SELECT COUNT(*) FROM conversations c
    LEFT JOIN statuses s ON c.id = s.conversation_id
    WHERE s.id IS NULL;
    ```

### Specifications

I am using Mastodon v3.4.1+glitch.

**Comment by ClearlyClaire** [comment]
Indeed, there seem to be no code to clean up empty conversations. We might want to add an index to be able to do that efficiently, as there is currently no index on `conversation_id` on `statuses`…
